packages/labs/analyzer/src/lib/paths.ts

```diff
@@ -0,0 +1,34 @@
+/**
+ * @license
+ * Copyright 2022 Google LLC
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+export type AbsolutePath = string & {
+  __absolutePathBrand: never;
+};
+
+/**
+ * A path relative to a package root
+ */
+export type PackagePath = string & {
+  __packagePathBrand: never;
+};
+
+export const absoluteToPackage = (
+  path: AbsolutePath,
+  root: AbsolutePath
```

Comment:
For clarity, prefer naming this `packageRoot` rather than `root`, which has a different meaning